### PR TITLE
Integration-ready image

### LINKED ISSUE #158: Fatal errors raised when run in production mode at request time
Steps to Reproduce:

Build Assets:

```sh
RAILS_ENV=production NODE_ENV=production rails assets:precompile
```

Run in production mode:

```sh
SECRET_KEY_BASE=MY_SECRET RAILS_ENV=production NODE_ENV=production rails server
```

so far so good. Now open http://localhost:3000 and check `log/production.log`:

```
I, [2018-01-15T20:10:18.314938 #42965]  INFO -- : [c6e2bb29-1faa-479c-9212-bbca8af3995e] Started GET "/" for 127.0.0.1 at 2018-01-15 20:10:18 -0800
F, [2018-01-15T20:10:18.315714 #42965] FATAL -- : [c6e2bb29-1faa-479c-9212-bbca8af3995e]   
F, [2018-01-15T20:10:18.319296 #42965] FATAL -- : [c6e2bb29-1faa-479c-9212-bbca8af3995e] NameError (uninitialized constant Infrastructure::CwdsAuthenticator::SecurityPolicy
Did you mean?  SecurityError):
F, [2018-01-15T20:10:18.319340 #42965] FATAL -- : [c6e2bb29-1faa-479c-9212-bbca8af3995e]   
F, [2018-01-15T20:10:18.319365 #42965] FATAL -- : [c6e2bb29-1faa-479c-9212-bbca8af3995e] lib/infrastructure/cwds_authenticator.rb:12:in `call'
I, [2018-01-15T20:10:18.503950 #42965]  INFO -- : [9732bec6-5918-41a6-8ddc-28ea69083fdb] Started GET "/favicon.ico" for 127.0.0.1 at 2018-01-15 20:10:18 -0800
F, [2018-01-15T20:10:18.504277 #42965] FATAL -- : [9732bec6-5918-41a6-8ddc-28ea69083fdb]   
F, [2018-01-15T20:10:18.507262 #42965] FATAL -- : [9732bec6-5918-41a6-8ddc-28ea69083fdb] NameError (uninitialized constant Infrastructure::CwdsAuthenticator::SecurityPolicy
Did you mean?  SecurityError):
F, [2018-01-15T20:10:18.507322 #42965] FATAL -- : [9732bec6-5918-41a6-8ddc-28ea69083fdb]   
F, [2018-01-15T20:10:18.507353 #42965] FATAL -- : [9732bec6-5918-41a6-8ddc-28ea69083fdb] lib/infrastructure/cwds_authenticator.rb:12:in `call'
```

### LINKED ISSUE #171: read service configs from ENV vars
integration deployment environments use ENV vars to declare service connections. We have to read those in and use them in the rails app.

### LINKED ISSUE #173: no style in production mode
screen shot 2018-01-19 at 4 16 59 pm

### PR DESCRIPTION
**Pull request by eckdanny-osi**
closes #171, closes #158, closes #173 

* [x] Read `CASE_API_BASE_URL` and `PERRY_API_BASE_URL` from env (#171 fix in 072a710)
* [x] Fix `eager_load_paths` (#158 fix in 5a94f27)
* [x] load stylesheets `bootstrap.css` and `application.css` (#173 fix in e52a951)
* [x] Fix `rspec` tests that be4e2f1 broke (fix in 5fc517e)
* [x] Fix rubocop warnings (fix in c2cd467)

### CODE REVIEW

**Comment by SashaKyz** [comment]
Can one of the admins verify this patch?

**Comment by eckdanny-osi** [comment]
5a94f27 closes #158

**Comment by eckdanny-osi** [comment]
In a separate PR, it'd be a good idea to update or create a `docker-compose[.whatever].yml` for running perry, api, redis, db2data, and casemanagement. IMO, the README could use an update how to run our image in general too.

Here's how i'm starting the container btw:

```bash
docker run -p 3001:3000 \
   -e RAILS_ENV=production \
   -e SECRET_KEY_BASE=MY_SECRET \
   -e RAILS_LOG_TO_STDOUT=TRUE \
   -e RAILS_SERVE_STATIC_FILES=TRUE \
   -e PERRY_API_BASE_URL=<some_url> \
   -e CASE_API_BASE_URL=<some_url> \
   cm_test
```

**Comment by eckdanny-osi** [comment]
Cool. the build went back to green on c2cd467. Removing **WiP** status

File: lib/infrastructure/cwds_authenticator.rb [hunk 1]
```diff
@@ -21,7 +21,8 @@ def redirect_to_login(url)
     end
 
     def login_url(callback)
-      "#{Rails.configuration.micro_services['perry_api_base_url']}/authn/login?callback=#{callback}"
+      perry_api_base_url = ENV.fetch('PERRY_API_BASE_URL', 'https://perry.test.cwds.io')
```
Comment (edgibbs) [inline lib/infrastructure/cwds_authenticator.rb]: I really don't want to fetch environment variables here, that's the reason for the microservices config, it's loaded once at runtime.

File: lib/infrastructure/security_gateway.rb [hunk 2]
```diff
@@ -10,7 +10,8 @@ def valid_token?(token)
     private
 
     def validation_url(token)
-      "#{Rails.configuration.micro_services['perry_api_base_url']}/authn/validate?token=#{token}"
+      perry_api_base_url = ENV.fetch('PERRY_API_BASE_URL', 'https://perry.test.cwds.io')
```
Comment (edgibbs) [inline lib/infrastructure/security_gateway.rb]: Same consideration here.

File: lib/infrastructure/http_service.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 module Infrastructure
   class HttpService
-    def initialize(base_url = Rails.configuration.micro_services['case_api_base_url'])
+    def initialize(base_url = ENV.fetch('CASE_API_BASE_URL', 'https://casemgmapi.test.cwds.io'))
```
Comment (edgibbs) [inline lib/infrastructure/http_service.rb]: And here.

File: lib/infrastructure/cwds_authenticator.rb (hunk 1 above)
Comment (eckdanny-osi) [reply lib/infrastructure/cwds_authenticator.rb]: It _was_ my intention to decouple config from rails. We need #171, #158, and #173 for #175. Agreed this solution needs some more thought, but with @edgibbs we agreed to address in a future PR as normal tech debt.

**Comment by edgibbs** [comment]
Will probably revisit the environment variables in the repositories..

File: lib/infrastructure/cwds_authenticator.rb (hunk 1 above)
Comment (amanbop) [reply lib/infrastructure/cwds_authenticator.rb]: @eckdanny-osi @edgibbs I agree with @edgibbs on this. Lets please resolve this.

**Pull request merged**